IPNS.md

```diff
@@ -29,95 +36,266 @@ Each time a file is modified, its content address changes. As a consequence, the
+- **Extensible Data** (DAG-CBOR)
+  - Extensible record data in [DAG-CBOR](https://ipld.io/specs/codecs/dag-cbor/spec/) format.
+  - The default set of fields can be augmented with additional information.
+    - Implementations are free to leverage this, or simply ignore unexpected fields.
+    - A good practice is to prefix custom field names with `_` to avoid
+      collisions with any new mandatory fields that may be added in a future
+      version of this specification.
+
+IPNS records are stored locally, as well as spread across the network, in order to be accessible to everyone.
+
+### Record Serialization Format
+
+For storing this structured data at rest and on the wire, we use `IpnsEntry` encoded as [protobuf](https://github.com/google/protobuf), which is a language-neutral, platform neutral extensible mechanism for serializing structured data.
+The extensible part of IPNS Record is placed in `IpnsEntry.data` field, which itself is encoded using a strict and deterministic subset of CBOR named [DAG-CBOR](https://ipld.io/specs/codecs/dag-cbor/spec/).
+
+```protobuf
 message IpnsEntry {
-	enum ValidityType {
-		// setting an EOL says "this record is valid until..."
-		EOL = 0;
-	}
-	required bytes value = 1;
-	required bytes signature = 2;
+ enum ValidityType {
+  // setting an EOL says "this record is valid until..."
+  EOL = 0;
+ }
+
+ // deserialized copy of data[value]
+ optional bytes value = 1;
+
+ // unused legacy field, use 'signatureV2' instead
+ optional bytes signature = 2;
+
+ // deserialized copies of data[validityType] and data[validity]
+ optional ValidityType validityType = 3;
+ optional bytes validity = 4;
+
+ // deserialized copy of data[sequence]
+ optional uint64 sequence = 5;
 
-	optional ValidityType validityType = 3;
-	optional bytes validity = 4;
+ // record TTL in nanoseconds, a deserialized copy of data[ttl]
+ optional uint64 ttl = 6;
 
-	optional uint64 sequence = 5;
+ // in order for nodes to properly validate a record upon receipt, they need the public
+ // key associated with it. For old RSA keys, its easiest if we just send this as part of
+ // the record itself. For newer Ed25519 keys, the public key can be embedded in the
+ // IPNS Name itself, making this field unnecessary.
+ optional bytes pubKey = 7;
 
-	optional uint64 ttl = 6;
+ // the signature of the IPNS record
+ optional bytes signatureV2 = 8;
 
-	optional bytes pubKey = 7;
+ // extensible record data in DAG-CBOR format
+ optional bytes data = 9;
 }
 ```
 
-## Protocol
+Notes:
 
-Taking into consideration a p2p network, each peer should be able to publish IPNS records to the network, as well as to resolve the IPNS records published by other peers.
+- IPNS implementations must support sending and receiving serialized
+  `IpnsEntry` of size less or equal 10 kiB. Handling records larger than 10 kiB
```

Comment:
Maybe "IPNS implementations must support sending and receiving a serialized `IpnsEntry` less than or equal to 10KiB."?

Reply:
Thanks. I applied your suggestions + created `Record Size Limit` section so we have single place with the value: https://github.com/ipfs/specs/pull/319/commits/1a8f0c908f7ada0a6164bdbfe76b1354355b8953